exasol/utils.py

```diff
@@ -17,3 +18,22 @@ def upward_file_search(file_name: str) -> str:
             return str(maybe_file)
         dir = dir.parent
     raise ValueError(f"Cannot find {file_name}")
+
+
+def optional_str_to_bool(value: Optional[str]) -> Optional[bool]:
+    """
+    Converts an optional string value to an optional boolean.
+    None, '' => None.
+    Case-insensitive "y", "yes", "true" => True
+    Case-insensitive "n", "no", "false" => False
+    Any other value cause a ValueError exception.
+    """
+    if value is None:
+        return None
+    value_l = value.lower()
+    if value_l in ["y", "yes", "true"]:
+        return True
+    elif value_l in ["n", "no", "false"]:
+        return False
+    else:
+        raise ValueError("Invalid boolean value " + value)
```

Comment:
```suggestion
def optional_str_to_bool(value: Optional[str]) -> Optional[bool]:
    """
    Converts an optional string value to an optional boolean.
    None, '' => None.
    Case-insensitive "y", "yes", "true" => True
    Case-insensitive "n", "no", "false" => False
    Any other value cause a ValueError exception.
    """
    if not value:
        return None

    mapping = {
        'y': True,
        'yes': True,
        'true': True,
        'n': False,
        'no': False,
        'false': False,
    }
    key = value.lower()

    try:
        result = mapping[key]
    except KeyError as ex:
        raise ValueError("Invalid boolean value " + value) from ex

    return result
```

Reply:
Maybe with fewer conditions, it will be easier to read and extend? (suggestion above)

Reply:
Thanks, yes, looks nicer this way. Also, there is a small bug in my code: an empty string will raise an exception, contrary to what the docstring says.